filebeat/harvester/log.go

```diff
@@ -291,40 +268,49 @@ func (h *Harvester) close() {
 	}
 }
 
-func createLineProcessor(
-	in source.FileSource,
-	codec encoding.Encoding,
-	bufferSize int,
-	maxBytes int,
-	readerConfig reader.LogFileReaderConfig,
-	jsonConfig *processor.JSONConfig,
-	mlrConfig *processor.MultilineConfig,
-	done chan struct{},
-) (processor.LineProcessor, error) {
+func (h *Harvester) newLogFileReaderConfig() reader.LogFileReaderConfig {
+	// TODO: NewLineReader uses additional buffering to deal with encoding and testing
+	//       for new lines in input stream. Simple 8-bit based encodings, or plain
+	//       don't require 'complicated' logic.
+	return reader.LogFileReaderConfig{
+		CloseRemoved:  h.Config.CloseRemoved,
+		CloseRenamed:  h.Config.CloseRenamed,
+		CloseOlder:    h.Config.CloseOlder,
+		CloseEOF:      h.Config.CloseEOF,
+		Backoff:       h.Config.Backoff,
+		MaxBackoff:    h.Config.MaxBackoff,
+		BackoffFactor: h.Config.BackoffFactor,
+	}
+}
+
+func (h *Harvester) newLineProcessor() (processor.LineProcessor, error) {
+
+	readerConfig := h.newLogFileReaderConfig()
+
 	var p processor.LineProcessor
 	var err error
 
-	fileReader, err := reader.NewLogFileReader(in, readerConfig, done)
+	fileReader, err := reader.NewLogFileReader(h.file, readerConfig, h.done)
```

Comment:
It could be that we continue reading so the offset is not at 0. So that means if we init the encoding too late, we have a problem here?

About setting the offset: I don't think that is a processors tasks, but the lines are still blury THB.